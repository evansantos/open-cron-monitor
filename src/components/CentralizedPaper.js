import styled from 'styled-components'

const CentralizedPaper = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`

export default CentralizedPaper
