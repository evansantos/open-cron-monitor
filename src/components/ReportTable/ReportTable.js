import React, { Component } from 'react'
import CentralizedPaper from '../CentralizedPaper'
import { Link } from 'react-router-dom'
import Paper from '@material-ui/core/Paper'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'


class ReportTable extends Component {
  render() {
    return (
      <CentralizedPaper>
        <Paper>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Status</TableCell>
                <TableCell>Name</TableCell>
                <TableCell>Started at</TableCell>
                <TableCell>Ended at</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {this.props.reports.map(report => (
                <TableRow key={report.id}>
                  <TableCell>{report.success ? 'Ok' : 'Error'}</TableCell>
                  <TableCell><Link to={`/report/${report.id}`}>{report.name}</Link></TableCell>
                  <TableCell>{report.startedAt}</TableCell>
                  <TableCell>{report.endedAt}</TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </Paper>
      </CentralizedPaper>
    )
  }
}

export default ReportTable
