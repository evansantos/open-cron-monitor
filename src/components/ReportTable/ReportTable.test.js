import Adapter from 'enzyme-adapter-react-16'
import Enzyme from 'enzyme'
import React from 'react'
import { shallow, mount, render } from 'enzyme'
import CentralizedPaper from '../CentralizedPaper'
import Paper from '@material-ui/core/Paper'
import ReportTable from './ReportTable'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'

Enzyme.configure({ adapter: new Adapter() })

describe('ReportTable', () => {
  it('should render properly', () => {
    const reports = [{ id: 1, success: true, name: 'name-1', createdAt: 'createdAt-1' }]
    const wrapper = shallow(<ReportTable reports={reports} />)

    expect(wrapper.find(CentralizedPaper).length).toBe(1)
    expect(wrapper.find(Paper).length).toBe(1)
    expect(wrapper.find(Table).length).toBe(1)
    expect(wrapper.find(TableBody).length).toBe(1)
    expect(wrapper.find(TableCell).length).toBe(6)
    expect(wrapper.find(TableHead).length).toBe(1)
    expect(wrapper.find(TableRow).length).toBe(2)
    expect(wrapper).toMatchSnapshot();
  })
})
