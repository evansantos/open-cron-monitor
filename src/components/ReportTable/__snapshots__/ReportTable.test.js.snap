// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTable should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportTable
    reports={
        Array [
            Object {
              "createdAt": "createdAt-1",
              "endedAt": "endedAt-1",
              "id": 1,
              "name": "name-1",
              "success": true,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Paper)>
        <WithStyles(Table)>
                <WithStyles(TableHead)>
                        <WithStyles(TableRow)>
                                <WithStyles(TableCell)>
                                        Status
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        Name
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        Started at
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        Ended at
                                </WithStyles(TableCell)>
                        </WithStyles(TableRow)>
                </WithStyles(TableHead)>
                <WithStyles(TableBody)>
                        <WithStyles(TableRow)>
                                <WithStyles(TableCell)>
                                        Ok
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        <Link
                                                replace={false}
                                                to="/report/1"
                                        >
                                                name-1
                                        </Link>
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        createdAt-1
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                        endedAt-1
                                </WithStyles(TableCell)>
                        </WithStyles(TableRow)>
                </WithStyles(TableBody)>
        </WithStyles(Table)>
</WithStyles(Paper)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Table)>
          <WithStyles(TableHead)>
                    <WithStyles(TableRow)>
                              <WithStyles(TableCell)>
                                        Status
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        Name
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        Started at
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        Ended at
                              </WithStyles(TableCell)>
                    </WithStyles(TableRow)>
          </WithStyles(TableHead)>
          <WithStyles(TableBody)>
                    <WithStyles(TableRow)>
                              <WithStyles(TableCell)>
                                        Ok
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        <Link
                                                  replace={false}
                                                  to="/report/1"
                                        >
                                                  name-1
                                        </Link>
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        createdAt-1
                              </WithStyles(TableCell)>
                              <WithStyles(TableCell)>
                                        endedAt-1
                              </WithStyles(TableCell)>
                    </WithStyles(TableRow)>
          </WithStyles(TableBody)>
</WithStyles(Table)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(TableHead)>
              <WithStyles(TableRow)>
                            <WithStyles(TableCell)>
                                          Status
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          Name
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          Started at
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          Ended at
                            </WithStyles(TableCell)>
              </WithStyles(TableRow)>
</WithStyles(TableHead)>,
            <WithStyles(TableBody)>
              <WithStyles(TableRow)>
                            <WithStyles(TableCell)>
                                          Ok
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          <Link
                                                        replace={false}
                                                        to="/report/1"
                                          >
                                                        name-1
                                          </Link>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          createdAt-1
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                                          endedAt-1
                            </WithStyles(TableCell)>
              </WithStyles(TableRow)>
</WithStyles(TableBody)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                                Status
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                                Name
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                                Started at
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                                Ended at
                </WithStyles(TableCell)>
</WithStyles(TableRow)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableCell)>
                    Status
</WithStyles(TableCell)>,
                  <WithStyles(TableCell)>
                    Name
</WithStyles(TableCell)>,
                  <WithStyles(TableCell)>
                    Started at
</WithStyles(TableCell)>,
                  <WithStyles(TableCell)>
                    Ended at
</WithStyles(TableCell)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Status",
                  },
                  "ref": null,
                  "rendered": "Status",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Name",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Started at",
                  },
                  "ref": null,
                  "rendered": "Started at",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Ended at",
                  },
                  "ref": null,
                  "rendered": "Ended at",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                                    Ok
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    <Link
                                                      replace={false}
                                                      to="/report/1"
                                    >
                                                      name-1
                                    </Link>
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    createdAt-1
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    endedAt-1
                  </WithStyles(TableCell)>
</WithStyles(TableRow)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(TableCell)>
                      Ok
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      <Link
                                            replace={false}
                                            to="/report/1"
                      >
                                            name-1
                      </Link>
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      createdAt-1
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      endedAt-1
</WithStyles(TableCell)>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Ok",
                    },
                    "ref": null,
                    "rendered": "Ok",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/report/1"
>
                        name-1
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "name-1",
                        "replace": false,
                        "to": "/report/1",
                      },
                      "ref": null,
                      "rendered": "name-1",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "createdAt-1",
                    },
                    "ref": null,
                    "rendered": "createdAt-1",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "endedAt-1",
                    },
                    "ref": null,
                    "rendered": "endedAt-1",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Paper)>
          <WithStyles(Table)>
                    <WithStyles(TableHead)>
                              <WithStyles(TableRow)>
                                        <WithStyles(TableCell)>
                                                  Status
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  Name
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  Started at
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  Ended at
                                        </WithStyles(TableCell)>
                              </WithStyles(TableRow)>
                    </WithStyles(TableHead)>
                    <WithStyles(TableBody)>
                              <WithStyles(TableRow)>
                                        <WithStyles(TableCell)>
                                                  Ok
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  <Link
                                                            replace={false}
                                                            to="/report/1"
                                                  >
                                                            name-1
                                                  </Link>
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  createdAt-1
                                        </WithStyles(TableCell)>
                                        <WithStyles(TableCell)>
                                                  endedAt-1
                                        </WithStyles(TableCell)>
                              </WithStyles(TableRow)>
                    </WithStyles(TableBody)>
          </WithStyles(Table)>
</WithStyles(Paper)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Table)>
            <WithStyles(TableHead)>
                        <WithStyles(TableRow)>
                                    <WithStyles(TableCell)>
                                                Status
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                Name
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                Started at
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                Ended at
                                    </WithStyles(TableCell)>
                        </WithStyles(TableRow)>
            </WithStyles(TableHead)>
            <WithStyles(TableBody)>
                        <WithStyles(TableRow)>
                                    <WithStyles(TableCell)>
                                                Ok
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                <Link
                                                            replace={false}
                                                            to="/report/1"
                                                >
                                                            name-1
                                                </Link>
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                createdAt-1
                                    </WithStyles(TableCell)>
                                    <WithStyles(TableCell)>
                                                endedAt-1
                                    </WithStyles(TableCell)>
                        </WithStyles(TableRow)>
            </WithStyles(TableBody)>
</WithStyles(Table)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(TableHead)>
                <WithStyles(TableRow)>
                                <WithStyles(TableCell)>
                                                Status
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                Name
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                Started at
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                Ended at
                                </WithStyles(TableCell)>
                </WithStyles(TableRow)>
</WithStyles(TableHead)>,
              <WithStyles(TableBody)>
                <WithStyles(TableRow)>
                                <WithStyles(TableCell)>
                                                Ok
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                <Link
                                                                replace={false}
                                                                to="/report/1"
                                                >
                                                                name-1
                                                </Link>
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                createdAt-1
                                </WithStyles(TableCell)>
                                <WithStyles(TableCell)>
                                                endedAt-1
                                </WithStyles(TableCell)>
                </WithStyles(TableRow)>
</WithStyles(TableBody)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                                    Status
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    Name
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    Started at
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                                    Ended at
                  </WithStyles(TableCell)>
</WithStyles(TableRow)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(TableCell)>
                      Status
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      Name
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      Started at
</WithStyles(TableCell)>,
                    <WithStyles(TableCell)>
                      Ended at
</WithStyles(TableCell)>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Status",
                    },
                    "ref": null,
                    "rendered": "Status",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Name",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Started at",
                    },
                    "ref": null,
                    "rendered": "Started at",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Ended at",
                    },
                    "ref": null,
                    "rendered": "Ended at",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                                        Ok
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                                        <Link
                                                            replace={false}
                                                            to="/report/1"
                                        >
                                                            name-1
                                        </Link>
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                                        createdAt-1
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                                        endedAt-1
                    </WithStyles(TableCell)>
</WithStyles(TableRow)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(TableCell)>
                        Ok
</WithStyles(TableCell)>,
                      <WithStyles(TableCell)>
                        <Link
                                                replace={false}
                                                to="/report/1"
                        >
                                                name-1
                        </Link>
</WithStyles(TableCell)>,
                      <WithStyles(TableCell)>
                        createdAt-1
</WithStyles(TableCell)>,
                      <WithStyles(TableCell)>
                        endedAt-1
</WithStyles(TableCell)>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Ok",
                      },
                      "ref": null,
                      "rendered": "Ok",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/report/1"
>
                          name-1
</Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "name-1",
                          "replace": false,
                          "to": "/report/1",
                        },
                        "ref": null,
                        "rendered": "name-1",
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "createdAt-1",
                      },
                      "ref": null,
                      "rendered": "createdAt-1",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "endedAt-1",
                      },
                      "ref": null,
                      "rendered": "endedAt-1",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
